packages/wdio-allure-reporter/src/utils.ts

```diff
@@ -186,5 +186,7 @@ export const setHistoryId = (test: AllureTest | undefined, suite: AllureGroup |
         .sort((a, b) => a.name?.localeCompare(b.name))
         .map(it => it.name + it.value)
         .join('')
-    test.historyId = md5(`${suite?.name}${test.wrappedItem.name}${paramsPart}`)
+    const hash = md5(`${suite?.name}${test.wrappedItem.name}${paramsPart}`)
+    test.historyId = hash
+    test.testCaseId = hash
```

Comment:
You have to check that testCaseId is empty before setting calculated one because user can set id via `allure.addAllureId(customId)` API and we should prefer this id over calculated one.

Reply:
turns out they are two different things, allure.addAllureId(customId) actually sets a label:
https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-allure-reporter/src/reporter.ts#L619

test.testCaseId is a set property so i can't even access it without some sort of breaking into private properties of AllureTest  object.

Reply:
tried to upload a file that has both testCaseId and explicitly set label via allure.addAllureId
allure.addAllureId actually overrides the generated id.
So... it kind of works, but i'm not sure if it's a sustainable thing.

i'll try to write a test in suite.test.ts where i can assert the final result of a report.

Reply:
>turns out they are two different things, allure.addAllureId(customId) actually sets a label:

It is the same thing.

Reply:
>So... it kind of works, but i'm not sure if it's a sustainable thing.

I prefer not to set generated id all if user provides his own. No matter how TestOps itself handles two id's (as field and as 
label) at the same time. If user set id, we need to send only user's id.

So please add guard that will set generated id only if no allureid label exists.

Reply:
got it